src/executors/SleepExecutor.js

```diff
@@ -1,7 +1,41 @@
 import ExecutorAbstract from '../ExecutorAbstract';
+import {sleep} from '../util/function';
 
 class SleepExecutor extends ExecutorAbstract {
+  //-----------------------------------
+  // Constructor
+  //-----------------------------------
+  /**
+   * Constructor for a new SleepExecutor, which will wait `milliseconds` before calling `this.done()`
+   *
+   * @param thread The parent thread that owns this executor.
+   * @param milliseconds The number of milliseconds to sleep
+   */
+  constructor(thread, milliseconds) {
+    super(thread);
 
+    this.milliseconds = milliseconds;
+  }
+
+  //-----------------------------------
+  // Methods
+  //-----------------------------------
+  /**
+   * Internal execution method called by Thread only.
+   *
+   * @param doneHandler The handler to call when done() is called.
+   * @param failHandler The handler to call when fail() is called;
+   * @private
+   */
+  _execute(doneHandler, failHandler) {
+    super._execute(doneHandler, failHandler);
+
+    sleep(this.milliseconds).then(() => this.done());
```

Comment:
Oooh, so `ExecutorAbstract` should be able to automagically handle promises returned from `_execute`. So you should be able to just do `return sleep(this.milliseconds)`.

Also, for next time, you could just do;

`sleep(this.milliseconds).then(this.done);`

Unless of course the promises has a result, then that result would be passed to done and done functions tend to treat any passed argument as an error (see Mocha, Jasmine, etc.)